### PR TITLE
Add support for pod subnet

### PR DESCRIPTION
**Pull request by mr-onion-2**
Add support for [pod_subnet_id](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#pod_subnet_id)

### CODE REVIEW

File: scripts [hunk 1]
```diff
@@ -0,0 +1 @@
+tfmod-scaffold/scripts
```
Comment (lonegunmanb) [inline scripts:1]: My bad, I should ignore this soft link in .gitignore file. Would you please remove this file from this pr? Thanks!

**Review by lonegunmanb — CHANGES_REQUESTED** [review]

File: scripts (hunk 1 above)
Comment (mr-onion-2) [reply scripts]: Done, sorry didn't notice that!

**Comment by lonegunmanb** [comment]
@mr-onion-2 sorry for the late reply, our test infrastructure was broken last week, and we've just fixed it. We've changed the test runner's pool name, would you please rebase this pr with the latest master branch so our new test pipeline can run the test? Apology for your inconveniency and thanks for your understanding.

**Review by lonegunmanb — APPROVED** [review]
Thanks @mr-onion-2 , LGTM!

**Pull request merged**